docs/src/main/docs/guides/05_Dockerfile.adoc

```diff
@@ -0,0 +1,302 @@
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+///////////////////////////////////////////////////////////////////////////////
+:adoc-dir: {guides-dir}
+
+= Creating Docker Images 
+:description: Helidon guide docker
+:keywords: helidon, guide, docker
+
+Building Docker images for Helidon applications.
+
+=== What You Will Learn
+
+You'll learn how to package your Helidon application and its runtime dependencies into a Docker
+image that also contains the Java runtime.
+
+=== What You Need
+
+|===
+|About 10 minutes
+| <<getting-started/01_prerequisites.adoc,Helidon Prerequisites>>
+|You'll also need Java 11 if you want to create custom JRE's using `jlink`
+|===
+
+=== Create Your Application Using The Helidon Quickstart
+
+Follow the instructions on the <<getting-started/02_base-example.adoc,Quickstart page>>
+to create a Helidon SE example. Once you've run the archetype to create
+the project come back here.
+
+=== Handling Runtime Dependencies
+
+Build the example:
+
+[source,bash,subs="verbatim,attributes"]
+----
+mvn clean package
+----
+
+When you run the maven build, you'll notice lines like the following:
+
+[source,bash,subs="verbatim,attributes"]
+----
+[INFO] Scanning for projects...
+. . .
+[INFO]
+[INFO] --- maven-dependency-plugin:2.9:copy-dependencies (copy-dependencies) @ quickstart-se ---
+[INFO] Copying netty-transport-4.1.22.Final.jar to /tmp/quickstart-se/target/libs/netty-transport-4.1.22.Final.jar
+. . .
+----
+
+The project uses the `maven-dependency-plugin` to copy the runtime dependencies to
+`target/libs/`. Additionally the `maven-jar-plugin` adds a `Class-Path` entry to the
+application's jar file so it can find those dependencies at runtime:
+
+[source,bash,subs="verbatim,attributes"]
+----
+unzip -p target/quickstart-se.jar META-INF/MANIFEST.MF 
+. . .
+Class-Path: libs/helidon-webserver-bundle-0.10.4.jar libs/helidon-webser
+ ver-0.10.4.jar libs/helidon-common-reactive-0.10.4.jar libs/helidon-com
+ mon-http-0.10.4.jar libs/helidon-common-key-util-0.10.4.jar libs/helido
+. . .
+----
+
+This means you can easily run the application jar:
+
+[source,bash,subs="verbatim,attributes"]
+----
+java -jar target/quickstart-se.jar
+----
+
+Now try the applicaion with `curl`:
+
+[source,bash,subs="verbatim,attributes"]
+----
+curl -X GET http://localhost:8080/greet
+{"message":"Hello World!"}
+----
+
+=== Creating a Docker Image
+
+Since the `target` directory has the runtime dependencies, the
+Dockerfile for our application is pretty simple. You can
+find the Dockerfile in the generated project at `src/main/docker/Dockerfile`.
+It should look like this:
+
+[source,yaml,subs="verbatim,attributes"]
+----
+FROM openjdk:8-jre-slim 
+
+RUN mkdir /app
+COPY libs /app/libs 
+COPY ${project.artifactId}.jar /app
+
+CMD ["java", "-jar", "/app/${project.artifactId}.jar"]
+----
+
+* We use a Java 8 JRE image provided by the OpenJDK project.
```

Comment:
I have not tried it with a `Dockerfile`, but you might try using AsciiDoc callouts in the `Dockerfile`? E.g., at the end of the `FROM` line add `# <1>` and at the end of the `RUN mkdir /app` line add `# <2>`. Then after the end of the AsciiDoc code block terminator (`----`) you can use 
```
<1> We use a Java 8 JRE image...
<2> We then create a directory...
```
This does mean editing the `Dockerfile` once it has been generated into the repository.

Reply:
Unfortunately the # for comments in a Dockerfile must be in column one, so that limits what you can do. I played with this and decided for this simple file I would not use them.